### DISCUSSION TITLE
Permament password changes on reboot

### DISCUSSION THREAD

**Discussion by Bolderaysky** [rustdesk/rustdesk#8666] [General]
### Bug Description

After setting permament password, I can connect to device as usual. However, after rebooting, permament password looks to be changed which prevents me from connecting to the device.

### How to Reproduce

1. Set permament password
2. Reboot
3. Try to connect

### Expected Behavior

1. Set permament password
2. Reboot
3. Connect successfully

### Operating system(s) on local side and remote side

Arch Linux -> Raspbian

### RustDesk Version(s) on local side and remote side

1.2.6 -> 1.2.6

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/73255292/829a6c68-973e-442e-9505-4d4b136768eb)

### Additional Context

I'm trying to connect to Raspberry pi 4B (Official Desktop flavor) from desktop PC powered by Linux. Usually, i don't have any issues like that, so it's possible it's something related to raspberry pi itself, but i'm not sure. 

Also, as mentioned in [this discussion](https://github.com/rustdesk/rustdesk/discussions/5003), I've checked if "password" length is different from what it should be. Before rebooting, I can connect and I can see that approximate length of password is same as I set, however, after rebooting, length becomes really large. Not sure if this is helpful though

Other features are working as intended, for example, Raspberry pi doesn't have any issues connecting to my private server after rebooting

I've installed Rustdesk using **aarch64 debian** package

**Comment by cucamorais** [comment]
Can confirm that this is happenning to me on 1.2.7 while using Win 10 on both sides.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/9232

**Comment by qicheng02** [comment]
me,too  ,  after  restart pc (win10)   the fixed password is  very Easy to be changed  and  the length of password  is  very long  ,
it has  a long time    and  it is Extremely distressed.

version: 1.33  1.34 1.35   will appear  .

**Comment by citydirector** [comment]
The same bug occurs on my device. The client version is 1.3.9, and the operating system is Windows 11 24h2. My permanent password is a 17 - character string that contains only upper - and lower - case letters and numbers.

**Reply by citydirector** [reply]
This is a relatively rare sporadic phenomenon, and I can't figure out where the cause lies. It has occurred twice in the past year or so.

**Comment by qicheng02** [comment]
This situation often occurs, and it is a very distressing situation
I personally believe that the password encryption part of the software itself may cause interference in some extreme cases, either by itself or by third-party programs
In either case, it is due to insufficient robustness of the program code itself

**Reply by citydirector** [reply]
So I have to use multiple remote control software at the same time.

**Comment by G325200** [comment]
Version 1.4.3 build 2025-10-17 17:21 Windows 10
Same issue happen to me. Password was 15 characters long, on boot password changed into 102 characters.

**Reply by G325200** [reply]
On connection attempt RustDesk.toml modified itself for whatever reason, at least from what I can see.

**Comment by wefalltomorrow** [comment]
Still happening to me as well :(

**Reply by 21pages** [reply]
If your password has become very long and you have not changed it, please try the nightly build:
https://github.com/rustdesk/rustdesk/releases/tag/nightly

The password may return to normal.

**Comment by rekloose** [comment]
I really want to use Rustdesk but this just keeps happening to me across multiple computers and I'm done.

**Comment by Jorilx** [comment]
Still happens on 1.4.9

**Reply by Jorilx** [reply]
I've finally found the source of my problem... I was missing `/etc/machine-id`, so dbus would create a different `/var/lib/dbus/machine-id` on each boot, so RustDesk couldn't decrypt the permanent password and generated a new temporary one.
